contracts/strategies/auraBAL/Strategy.sol

```diff
@@ -0,0 +1,202 @@
+import "../../../interfaces/balancer/IRewardHandler.sol";
+import "./StrategyBase.sol";
+
+contract AuraBalStrategy is Ownable, AuraBalStrategyBase {
+    using SafeERC20 for IERC20;
+
+    address public immutable vault;
+    address[] public rewardTokens;
+    mapping(address => address) public rewardHandlers;
+
+    uint256 public constant FEE_DENOMINATOR = 10000;
+
+    constructor(address _vault) {
+        vault = _vault;
+    }
+
+    /// @notice Set approvals for the contracts used when swapping & staking
+    function setApprovals() external {
+        IERC20(AURABAL_TOKEN).safeApprove(AURABAL_STAKING, 0);
+        IERC20(AURABAL_TOKEN).safeApprove(AURABAL_STAKING, type(uint256).max);
+        IERC20(BAL_TOKEN).safeApprove(BAL_VAULT, 0);
+        IERC20(BAL_TOKEN).safeApprove(BAL_VAULT, type(uint256).max);
+        IERC20(WETH_TOKEN).safeApprove(BAL_VAULT, 0);
+        IERC20(WETH_TOKEN).safeApprove(BAL_VAULT, type(uint256).max);
+        IERC20(BAL_ETH_POOL_TOKEN).safeApprove(AURABAL_PT_DEPOSIT, 0);
+        IERC20(BAL_ETH_POOL_TOKEN).safeApprove(
+            AURABAL_PT_DEPOSIT,
+            type(uint256).max
+        );
+        IERC20(BAL_ETH_POOL_TOKEN).safeApprove(BAL_VAULT, 0);
+        IERC20(BAL_ETH_POOL_TOKEN).safeApprove(BAL_VAULT, type(uint256).max);
+    }
+
+    /// @notice update the token to handler mapping
+    function _updateRewardToken(address _token, address _handler) internal {
+        rewardHandlers[_token] = _handler;
+    }
+
+    /// @notice Add a reward token and its handler
+    /// @dev For tokens that should not be swapped (i.e. BAL rewards)
+    ///      use address as zero handler
+    /// @param _token the reward token to add
+    /// @param _handler address of the contract that will sell for BAL or ETH
+    function addRewardToken(address _token, address _handler)
+        external
+        onlyOwner
+    {
+        rewardTokens.push(_token);
+        _updateRewardToken(_token, _handler);
+    }
+
+    /// @notice Update the handler of a reward token
+    /// @dev Used to update a handler or retire a token (set handler to address 0)
+    /// @param _token the reward token to add
+    /// @param _handler address of the contract that will sell for BAL or ETH
+    function updateRewardToken(address _token, address _handler)
+        external
+        onlyOwner
+    {
+        _updateRewardToken(_token, _handler);
+    }
+
+    /// @notice returns the number of reward tokens
+    /// @return the number of reward tokens
+    function totalRewardTokens() external view returns (uint256) {
+        return rewardTokens.length;
+    }
+
+    /// @notice Query the amount currently staked
+    /// @return total - the total amount of tokens staked
+    function totalUnderlying() public view returns (uint256 total) {
+        return auraBalStaking.balanceOf(address(this));
+    }
+
+    /// @notice Deposits underlying tokens in the staking contract
+    function stake(uint256 _amount) external onlyVault {
```

Comment:
You're not using this function in `Strategy.sol` except for your tests. In the `harvest()` function at the end you directly call `auraBalStaking.stake(_stakingAmount);` instead. Not sure if you want to remove the stake function or replace the final `auraBalStaking.stake(_stakingAmount);` with a `stake()` one.

Edit: I've noticed this for `withdraw()` as well. If there's an external call from the vault somehwere else I'm missing you can ignore this remark.

Edit: nvm, found it: `IStrategy(strategy).stake(_amount);` in GenericVault.sol.